operators/src/processing/rasterization/mod.rs

```diff
@@ -0,0 +1,1089 @@
+use crate::engine::TypedVectorQueryProcessor::MultiPoint;
+use crate::engine::{
+    CreateSpan, ExecutionContext, InitializedRasterOperator, InitializedVectorOperator, Operator,
+    OperatorName, QueryContext, QueryProcessor, RasterOperator, RasterQueryProcessor,
+    RasterResultDescriptor, SingleVectorSource, TypedRasterQueryProcessor,
+    TypedVectorQueryProcessor,
+};
+use arrow::datatypes::ArrowNativeTypeOp;
+use std::sync::{Arc, Mutex};
+
+use crate::error;
+use crate::processing::rasterization::GridOrDensity::Grid;
+use crate::util;
+
+use async_trait::async_trait;
+
+use futures::stream::BoxStream;
+use futures::{stream, StreamExt};
+use geoengine_datatypes::collections::GeometryCollection;
+
+use geoengine_datatypes::primitives::{
+    AxisAlignedRectangle, BoundingBox2D, Coordinate2D, Measurement, RasterQueryRectangle,
+    SpatialPartition2D, SpatialPartitioned, SpatialResolution, VectorQueryRectangle,
+};
+use geoengine_datatypes::raster::{
+    GeoTransform, Grid2D, GridOrEmpty, GridSize, GridSpaceToLinearSpace, RasterDataType,
+    RasterTile2D, TilingSpecification,
+};
+
+use num_traits::FloatConst;
+use rayon::prelude::*;
+
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+
+use crate::util::{spawn_blocking, spawn_blocking_with_thread_pool};
+use tracing::span;
+use tracing::Level;
+use typetag::serde;
+
+/// An operator that rasterizes vector data
+pub type Rasterization = Operator<RasterizationParams, SingleVectorSource>;
+
+impl OperatorName for Rasterization {
+    const TYPE_NAME: &'static str = "Rasterization";
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum GridSizeMode {
+    /// The spatial resolution is interpreted as a fixed size in coordinate units
+    Fixed,
+    /// The spatial resolution is interpreted as a multiplier for the query pixel size
+    Relative,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "type")]
+pub enum GridOrDensity {
+    /// A grid which summarizes points in cells (2D histogram)
+    Grid(GridParams),
+    /// A heatmap calculated from a gaussian density function
+    Density(DensityParams),
+}
+
```

Comment:
It is allowed to have a spatial resolution of (0,0). Maybe we should prohibit that?

Reply:
This is one of the reasons I limited the grid resolution to the query resolution (= its size to the tile size). I assumed that there is already some kind of check in place which prevents the user from ever querying a (0,0) resolution, which would also lead to all kinds of problems elsewhere (e.g., tiling strategy).
If you want me to prohibit it here I see no disadvantage, though. The one use case I could see is if someone wants the grid resolution to become "infinitely" smaller when zooming in. However, the same can be achieved with the relative grid size mode and a multiplier of 1.